src/routes/users.routes.ts

```diff
@@ -11,20 +11,37 @@ const SALT_ROUNDS = 10
 
 export async function userRoutes(app: FastifyTypeInstance) {
     app.get('/users', {
-        schema: { // esquema que aparecera na documentacao
-            tags: ['users'], // tag para agrupar requests para users
-            description: 'List users', // descricao do que o metodo faz nessa rota
-            response: { // estruturas de respostas que a rota entrega para cada http code
-                200: z.array(z.object({
-                    id: z.string(),
-                    name: z.string(),
-                    email: z.string(),
-                }))
-            }
+      schema: {
+        tags: ['users'],
+        description: 'List users',
+        response: {
+          200: z.array(userSchema.omit({ password: true })),
         }
+      }
     }, () => {
-        return usersTableSim
-    })
+      return usersTableSim.filter(u => !u.deleted).map(({ password, ...rest }) => rest); //soft delete
+    });
+
+    app.get('/users/:id', {
+        schema: {
+            tags: ['users'],
+            description: 'Get an user by ID',
+            response: {
+                200: userSchema.omit({ password: true}),
+                404: z.object({ message: z.string() }).describe('User not found'),
+            },
+        }
+    }, (req, rep) => {
+        const {id} = req.param as {id: string};
```

Comment:
should add `params` at the schema
```suggestion
        const {id} = req.params as {id: string};
```